inc/Remote/ApiCore.php

```diff
@@ -582,6 +586,76 @@ public function appendPage($id, $text, $params = array())
         return $this->putPage($id, $currentpage . $text, $params);
     }
 
+    /**
+     * Create one or more users
+     *
+     * @param array[] $userStruct User struct
+     *
+     * @return boolean Create state
+     *
+     * @throws AccessDeniedException
+     * @throws RemoteException
+     */
+    public function createUser($userStruct)
+    {
+        if (!auth_isadmin()) {
+            throw new AccessDeniedException('Only admins are allowed to create users', 114);
+        }
+
+        /** @var \dokuwiki\Extension\AuthPlugin $auth */
+        global $auth;
+
+        if(!$auth->canDo('addUser')) {
+            throw new AccessDeniedException(
+                sprintf('Authentication backend %s can\'t do addUser', $auth->getPluginName()),
+                114
+            );
+        }
+
+        $user = trim($auth->cleanUser($userStruct['user'] ?? ''));
+        $password = $userStruct['password'] ?? '';
+        $name = trim(preg_replace('/[\x00-\x1f:<>&%,;]+/', '', $userStruct['name'] ?? ''));
+        $mail = trim(preg_replace('/[\x00-\x1f:<>&%,;]+/', '', $userStruct['mail'] ?? ''));
```

Comment:
The cleanup you do here using trim and preg_replace - is this referencing any similar code used elsewhere in DokuWiki? I don't think the remoteAPI should behave any different than what we usually do when registering a user.

Reply:
This is already used in the register function:
https://github.com/splitbrain/dokuwiki/blob/6dd998877521c56871ba43e97f2fa2590a9a69cf/inc/auth.php#L870-L932

Reply:
I couldn't access the register function as it refers to input vars. I wanted the result be the same no matter if the user was supplied by the xmlapi or via form. If I defined the xmlapi that way the api client has to check all data before submit I would have to transport that logic into every api client application in use.

Reply:
It was introduced in 54f0e6ea83591f920ac7b4719213e38f317740ed to fix https://bugs.dokuwiki.org/960.html

I guess I would prefer if this would be all encapsulated in one method/class that gets it's input as parameters and then is used in handling registration from the web interface and the API. But that's beyond the scope of this PR so I'm fine with leaving it as is for now. In a future refactoring it might make sense to move this whole functionality to the Register action (there's a FIXME in there already) and have the API make a call on that.